### ISSUE TITLE
cell-relax with `fixed_axes c` of 2D material finally carried out the `ab` vectors not in `xy` plane

### ISSUE THREAD

**Issue by xdzhu** [deepmodeling/abacus-develop#3751]
### Describe the bug

When using ABACUS for cell-relax, the `fixed_axes c` option was used, and as a result, both lattice vectors **_a_** and **_b_** have components in the **_z_**-direction. How to only relax the **_xy_**-components of _**ab**_ vectors with the z-component constrained?
Run ABACUS cell-relax with `fixed_axes c` :
![image](https://github.com/deepmodeling/abacus-develop/assets/55775434/b74c3412-714f-4443-b362-d3aadde3bbf1)

The constrain_z implementation is critical in 2D material calculation. I wonder whether ABACUS can provide a similar implementation to VASP's constrain_z.
Run VASP cell-relax with constrain_Z compiled vasp_std:
![image](https://github.com/deepmodeling/abacus-develop/assets/55775434/f5614c82-06d2-4837-ba5a-ae8a9756b08d)

You can check the result in as below:
[fixed_c-VS_constarin_z.zip](https://github.com/deepmodeling/abacus-develop/files/14662528/fixed_c-VS_constarin_z.zip)

### Expected behavior

More methods in Constrained Optimization

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

**Comment by WHUweiqingzhou** [comment]
It seems that ABACUS lacks an option for fixed lattice angle.

**Comment by xdzhu** [comment]
> It seems that ABACUS lacks an option for fixed lattice angle.

That‘s right. Need more fixed options like: alpha, beta, gamma. OR, just provide a specific way to only relax the in-plane lattice parametes of a slab model?

**Comment by mohanchen** [comment]
Reopen it if anyone needs
